app/code/Magento/Wishlist/Model/Wishlist.php

```diff
@@ -334,7 +334,7 @@ protected function _addCatalogProduct(Product $product, $qty = 1, $forciblySetQt
     {
         $item = null;
         foreach ($this->getItemCollection() as $_item) {
-            if ($_item->representProduct($product)) {
+            if ($_item->getProduct()->getId() === $product->getId()) {
```

Comment:
I would not recommend going this way since this fix excludes an important comparison of item options. See the following implementation for details 

https://github.com/magento/magento2/blob/795b2c6152e0fdf27bfb720d18e1bcd29723b154/app/code/Magento/Wishlist/Model/Item.php#L585

By bypassing this check we introduce a new problem when items with different options might be the same product.

Also, if the issue appears only in GraphQL are, I would recommend checking and fixing the issue on the resolvers or GraphQL services level without modifying the domain logic.

Hope it makes sense.

Reply:
@ranaprathapm-kensium the above comment makes sense. This issue should be fixed in the graphql area. Also please cover this case using web api test. Thanks!